authority/provisioner/webhook.go

```diff
@@ -0,0 +1,204 @@
+package provisioner
+
+import (
+	"bytes"
+	"context"
+	"crypto/hmac"
+	"crypto/sha256"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"text/template"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/smallstep/certificates/templates"
+	"go.step.sm/crypto/sshutil"
+	"golang.org/x/crypto/ssh"
+)
+
+type Webhook struct {
+	ID                   string `json:"id"`
+	Name                 string `json:"name"`
+	URL                  string `json:"url"`
+	Kind                 string `json:"kind"`
+	DisableTLSClientAuth bool   `json:"disableTLSClientAuth,omitempty"`
+	SigningSecret        string `json:"-"`
+	BearerToken          string `json:"-"`
+	BasicAuth            struct {
+		Username string
+		Password string
+	} `json:"-"`
+}
+
+type sshRequest struct {
+	Key        []byte
+	Type       string
+	KeyID      string
+	Principals []string
+}
+
+// ssh.Certificate but the Key and SignatureKey are marshaled
+type sshCert struct {
+	Nonce    []byte
+	Key      []byte
+	Serial   uint64
+	CertType uint32
+	//nolint:revive // field name matches ssh.Certificate
+	KeyId           string
+	ValidPrincipals []string
+	ValidAfter      uint64
+	ValidBefore     uint64
+	Permissions     ssh.Permissions
+	Reserved        []byte
+	SignatureKey    []byte
+	Signature       *ssh.Signature
+}
```

Comment:
you probably want JSON tags here. And you can just fix that revive error; at the end of the day, it doesn't matter if they don't perfectly match.

Reply:
d976e88